src/forms/registration-popup/index.jsx

```diff
@@ -107,20 +108,17 @@ const RegistrationForm = () => {
   ]);
 
   useEffect(() => {
-    const focusInterval = setInterval(() => {
-      if (socialAuthnButtonRef.current) {
+    if (thirdPartyAuthApiStatus === COMPLETE_STATE) {
+      if (providers.length > 0) {
         socialAuthnButtonRef.current.focus();
-        clearInterval(focusInterval);
       } else if (emailRef.current) {
         emailRef.current.focus();
       }
-    }, 100);
-
-    return () => clearInterval(focusInterval);
-  }, []);
+    }
+  }, [thirdPartyAuthApiStatus, providers]);
 
   useEffect(() => {
-    if (errors && Object.keys(errors).length > 0 && errorRef.current) {
+    if (Object.keys(errors).length > 0 && Object.values(errors)[0]) {
```

Comment:
Can we improve `Object.values(errors)[0]` condition? if providers don't exist the focus will be on the error alert instead of the email field.

Reply:
Yes, the condition doing the same,  if providers don't exist, the focus will be on the error alert instead of the email field.
<img width="1175" alt="Screenshot 2024-06-11 at 2 08 45 PM" src="https://github.com/edx/frontend-component-authn-edx/assets/7627421/2ac08a26-3828-4c79-9c71-576e08516fe7">

Reply:
What would be the behavior on first time component render without clicking the submit button.

Reply:
The focus will be on the Email field.
<img width="634" alt="Screenshot 2024-06-11 at 2 33 46 PM" src="https://github.com/edx/frontend-component-authn-edx/assets/7627421/d57f60b0-4dce-4741-8c88-198152815ecc">